### PR TITLE
Sparse CSR CPU: implement addmm(dense, sparse, sparse) -> dense

### PR DESCRIPTION
**Pull request by IvanYashchuk**
This PR adds a possibility to multiply two sparse matrices and add the result of a product to a dense matrix.
It uses [MKL spmmd function](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/blas-and-sparse-blas-routines/inspector-executor-sparse-blas-routines/inspector-executor-sparse-blas-execution-routines/mkl-sparse-spmmd.html) and only CPU path is implemented for now.

Ref. https://github.com/pytorch/pytorch/issues/60858

### CODE REVIEW

**Review by cpuhrsch — APPROVED** [review]

File: aten/src/ATen/native/mkl/SparseBlasImpl.cpp [hunk 1]
```diff
@@ -206,6 +206,58 @@ void addmm_dense_result(
...
+            mkl_A.descriptor(),
+            mkl_B.descriptor(),
+            order,
+            C_.data_ptr<scalar_t>(),
+            ldc);
+      });
+
+  // If beta is zero NaN and Inf should not be propagated to the result
+  if (beta.toComplexDouble() == 0.) {
+    C.zero_();
```
[the remark this answers is not in the document]
Comment (IvanYashchuk) [reply aten/src/ATen/native/mkl/SparseBlasImpl.cpp:253]: Yes. `addmm` computes alpha * (A @ B) + beta * C, if beta is zero then the result would be equal only to A @ B portion computed by MKL.

**Comment by cpuhrsch** [comment]
@IvanYashchuk - the windows errors might be unrelated. Could you try rebasing?

**Comment by IvanYashchuk** [comment]
Let's see whether the build succeeds now.

**Comment by cpuhrsch** [comment]
@IvanYashchuk - they're failing again. This might be because of this PR.

**Comment by IvanYashchuk** [comment]
> @IvanYashchuk - they're failing again. This might be because of this PR.

Thanks for the ping! Will take a look and fix it soon.

**Comment by cpuhrsch** [comment]
I found
```
2022-02-23T11:09:46.5415921Z caffe2\CMakeFiles\torch_cpu.dir\__\aten\src\ATen\native\cpu\Activation.cpp.AVX512.cpp.obj lib\breakpad.lib win_tmp\mkl\lib\libiomp5md.lib win_tmp\mkl\lib\libiomp5md.lib lib\c10.lib lib\pthreadpool.lib lib\cpuinfo.lib lib\XNNPACK.lib lib\fbgemm.lib lib\gloo.lib lib\foxi_loader.lib lib\kineto.lib win_tmp\mkl\lib\mkl_intel_lp64.lib win_tmp\mkl\lib\mkl_intel_thread.lib win_tmp\mkl\lib\mkl_core.lib win_tmp\mkl\lib\libiomp5md.lib win_tmp\mkl\lib\mkl_intel_lp64.lib win_tmp\mkl\lib\mkl_intel_thread.lib win_tmp\mkl\lib\mkl_core.lib win_tmp\mkl\lib\libiomp5md.lib win_tmp\mkl\lib\mkl_intel_lp64.lib win_tmp\mkl\lib\mkl_intel_thread.lib win_tmp\mkl\lib\mkl_core.lib win_tmp\mkl\lib\libiomp5md.lib lib\dnnl.lib lib\cpuinfo.lib lib\breakpad_common.lib Imagehlp.lib lib\caffe2_protos.lib lib\pthreadpool.lib lib\clog.lib lib\asmjit.lib C:\Jenkins\Miniconda3\Library\lib\uv.lib "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.3\lib\x64\cudart.lib" lib\onnx.lib lib\onnx_proto.lib lib\libprotobuf.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib 
2022-02-23T11:09:46.5419895Z mkl_core.lib(_avx512_jit_destroy.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in AccumulateType.cpp.obj
2022-02-23T11:09:46.5421331Z mkl_core.lib(_avx2_jit_destroy.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in AccumulateType.cpp.obj
2022-02-23T11:09:46.5422756Z mkl_core.lib(_avx_jit_destroy.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MT_StaticRelease' doesn't match value 'MD_DynamicRelease' in AccumulateType.cpp.obj
2022-02-23T11:09:46.5423736Z    Creating library lib\torch_cpu.lib and object lib\torch_cpu.exp
2022-02-23T11:09:46.5424312Z bin\torch_cpu.dll : fatal error LNK1319: 3 mismatches detected
```
in the [raw logs](https://pipelines.actions.githubusercontent.com/mBh68xKhi8LyM7tp3vECvYXNFvuV4gyVGgmYCteuEZP9JH92QN/_apis/pipelines/1/runs/1786409/signedlogcontent/2?urlExpires=2022-02-23T16%3A56%3A42.7998598Z&urlSigningMethod=HMACV1&urlSignature=%2BVwYk98ptnfTehipMeFg1qc9amCHA7E%2BGzOUf3gwVAU%3D) of [win-vs2019-cuda11.3-py3](https://github.com/pytorch/pytorch/runs/5301494985?check_suite_focus=true).

**Pull request closed without merging**